.github/workflows/pytest.yml

```diff
@@ -191,14 +197,14 @@ jobs:
           mv .github/.coveragerc .
 
       - name: Download all artifacts
-        uses: actions/download-artifact@eaceaf801fd36c7dee90939fad912460b18a1ffe # v4
+        uses: actions/download-artifact@87c55149d96e628cc2ef7e6fc2aab372015aec85 # v4
```

Comment:
Just out of curiosity, why in some instances the hash has been substituted by `v#` while here for instance is kept?

Reply:
we pin to specific commits because people can do ugly things with github actions...but we trust actions under our control (e.g. setup-nextflow) so we can be a bit more relaxed with the version pinning there (plus our actions are not as rigorously testes/have a smaller user size, so breaking bugs creep in more easily)